### PR TITLE
Use M1 runner to build arm64 macOS builds

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
# Issue

GitHub now provides M1 based runner, so we can use it https://github.com/actions/runner-images/issues/9254

Besides that I had to update Boost to version 1.85.0, because otherwise we had compilation errors on this new macOS.

## Tasklist

 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by DennisOSRM** [comment]
Nice, looks like this is much cleaner than #6864.

**Comment by SiarheiFedartsou** [comment]
> Nice, looks like this is much cleaner than #6864.

Ah sorry, I haven’t noticed your PR when started working on this…

File: .github/workflows/osrm-backend.yml [hunk 1]
```diff
@@ -331,7 +331,7 @@ jobs:
             build_node_package: true
             continue-on-error: true
             node: 20
-            runs-on: macos-11
+            runs-on: macos-13 # x86_64
```
Comment (SiarheiFedartsou) [inline .github/workflows/osrm-backend.yml:334]: Looking at  https://github.com/Project-OSRM/osrm-backend/pull/6864 it seems we can use macos-14-large here, right @DennisOSRM ?

**Comment by DennisOSRM** [comment]
> > Nice, looks like this is much cleaner than #6864.
> 
> 
> 
> Ah sorry, I haven’t noticed your PR when started working on this…

No worries. Great minds think alike. 😉

File: .github/workflows/osrm-backend.yml (hunk 1 above)
Comment (DennisOSRM) [reply .github/workflows/osrm-backend.yml:334]: Yes, that should probably work.

**Review by DennisOSRM — APPROVED** [review]
👋🏼

**Pull request merged**